akka-stream/src/main/scala/akka/stream/impl/UnfoldResourceSourceAsync.scala

```diff
@@ -107,11 +107,12 @@ import scala.util.control.NonFatal
 
     private def createResource(): Unit = {
       create().onComplete { resource =>
-        createdCallback(resource).recover {
+        createdCallback(resource).failed.foreach {
           case _: StreamDetachedException =>
             // stream stopped
             resource match {
-              case Success(r)  => close(r)
+              case Success(r) =>
+                close(r)
               case Failure(ex) => throw ex // failed to open but stream is stopped already
```

Comment:
where does this exception go? do we care about it?

Reply:
Default exception handler on the dispatcher threadpool?